lldb/tools/lldb-dap/Handler/ThreadsRequestHandler.cpp

```diff
@@ -50,16 +50,26 @@ namespace lldb_dap {
 // }
 void ThreadsRequestHandler::operator()(
     const llvm::json::Object &request) const {
-  lldb::SBProcess process = dap.target.GetProcess();
   llvm::json::Object response;
   FillResponse(request, response);
 
-  const uint32_t num_threads = process.GetNumThreads();
   llvm::json::Array threads;
-  for (uint32_t thread_idx = 0; thread_idx < num_threads; ++thread_idx) {
-    lldb::SBThread thread = process.GetThreadAtIndex(thread_idx);
-    threads.emplace_back(CreateThread(thread, dap.thread_format));
+  // Client requests the baseline of currently existing threads after
+  // a successful launch or attach.
+  // A 'threads' request will be sent after configurationDone response.
+  // Return the cache of initial threads as process might have resumed
+  if (dap.initial_thread_list) {
+    threads = dap.initial_thread_list.value();
+    dap.initial_thread_list.reset();
+  } else {
+    lldb::SBProcess process = dap.target.GetProcess();
+    const uint32_t num_threads = process.GetNumThreads();
+    for (uint32_t thread_idx = 0; thread_idx < num_threads; ++thread_idx) {
+      lldb::SBThread thread = process.GetThreadAtIndex(thread_idx);
+      threads.emplace_back(CreateThread(thread, dap.thread_format));
+    }
```

Comment:
Use JSONUtils.cpp function `GetThreads`:
```
threads = GetThreads(dap.target.GetProcess());
```
This ensures if we update how threads are fetched that we don't have two sections of code doing the same thing.